lib/edit-site-page.php

```diff
@@ -225,3 +213,33 @@ function gutenberg_edit_site_init( $hook ) {
 	wp_enqueue_style( 'wp-format-library' );
 }
 add_action( 'admin_enqueue_scripts', 'gutenberg_edit_site_init' );
+
+/**
+ * Register a core site setting for front page information.
+ */
+function register_site_editor_homepage_settings() {
+	register_setting(
+		'general',
+		'show_on_front',
+		array(
+			'show_in_rest' => array(
+				'name' => 'show_on_front',
+			),
+			'type'         => 'string',
+			'description'  => __( 'What to show on the front page', 'gutenberg' ),
+		)
+	);
+
+	register_setting(
+		'general',
+		'page_on_front',
+		array(
+			'show_in_rest' => array(
+				'name' => 'page_on_front',
+			),
+			'type'         => 'number',
+			'description'  => __( 'The ID of the page that should be displayed on the front page', 'gutenberg' ),
+		)
+	);
+}
+add_action( 'rest_api_init', 'register_site_editor_homepage_settings', 10 );
```

Comment:
I believe this is adding data to some REST API endpoint. So pinging @TimothyBJacobs for awareness.

Reply:
Thanks for the ping! Doing this on `rest_api_init` should work, but it might be better to do it on `init` so it is always a registered setting. We can also safely drop the `name` parameter for `show_in_rest`. It'll default to the option name.

Reply:
Nice, I will definitely make those changes.

Reply:
https://github.com/WordPress/gutenberg/pull/24390